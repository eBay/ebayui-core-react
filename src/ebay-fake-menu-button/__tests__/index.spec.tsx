import React from 'react'
import { fireEvent, render } from '@testing-library/react';
import initStoryshots from '@storybook/addon-storyshots'
import { EbayFakeMenuButton, EbayFakeMenuButtonItem } from '..'

initStoryshots({
    config: ({ configure }) => configure(() => require('./index.stories.tsx'), module)
})

jest.useFakeTimers()

describe('<EbayFakeMenuButton>', () => {
    describe('on button click', () => {
        let spy
        beforeEach(() => {
            spy = jest.fn()
        })
        it('should fire onExpand event', () => {
            const wrapper = render(
                <EbayFakeMenuButton onExpand={spy}>
                    <EbayFakeMenuButtonItem />
                </EbayFakeMenuButton>
            )
            fireEvent.click(wrapper.getByRole('button'))

            expect(spy).toBeCalled()
        })
        it('should fire onCollapse event', () => {
            const wrapper = render(
                <EbayFakeMenuButton onCollapse={spy}>
                    <EbayFakeMenuButton />
                </EbayFakeMenuButton>
            )

            const button = wrapper.getByRole('button')
            fireEvent.click(button)
            fireEvent.click(button)

            expect(spy).toBeCalled()
        })
    })
    describe('on opened menu', () => {
        let spy, button
        beforeEach(() => {
            spy = jest.fn()
            const wrapper = render(
                <EbayFakeMenuButton onCollapse={spy}>
                    <EbayFakeMenuButtonItem />
                </EbayFakeMenuButton>
            )
            button = wrapper.getByRole('button')
            fireEvent.click(button)
        })

        it('should close on button click', () => {
            fireEvent.click(button)

            expect(spy).toBeCalled()
        })

        // todo: fix these tests
        // it('should close on Esc press', () => {
        //     fireEvent.keyDown(menu, { key: 'Escape' })
        //
        //     expect(spy).toBeCalled()
        // })
        //
        // it('should close on BG click', () => {
        //     jest.runAllTimers()
        //     document.body.click();
        //
        //     expect(spy).toBeCalled()
        // })
    })
})
