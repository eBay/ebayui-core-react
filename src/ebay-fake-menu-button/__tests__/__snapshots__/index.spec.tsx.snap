// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-fake-menu-button Borderless 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--borderless"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        eBay Menu without borders!
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-fake-menu-button Default 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        eBay Menu
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-fake-menu-button Disabled 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    disabled={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        eBay Menu
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-fake-menu-button Expanded 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={true}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        eBay Menu
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <div
    className="fake-menu-button__menu fake-menu"
    tabIndex={-1}
  >
    <ul
      className="fake-menu__items"
      tabIndex={-1}
    >
      <li>
        <a
          className="fake-menu-button__item fake-menu__item"
          href="http://ebay.com"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span>
            item 1 that has very long text
          </span>
          <svg
            aria-hidden={true}
            className="icon icon--tick-small"
            focusable={false}
            height="16px"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlinkHref="#icon-tick-small"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          className="fake-menu-button__item fake-menu__item"
          href="http://ebay.com"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span>
            item 2
          </span>
          <svg
            aria-hidden={true}
            className="icon icon--tick-small"
            focusable={false}
            height="16px"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlinkHref="#icon-tick-small"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          className="fake-menu-button__item fake-menu__item"
          href="http://ebay.com"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span>
            item 3
          </span>
          <svg
            aria-hidden={true}
            className="icon icon--tick-small"
            focusable={false}
            height="16px"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlinkHref="#icon-tick-small"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</span>
`;

exports[`Storyshots ebay-fake-menu-button Fixed Width 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        Menu has a button width
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-fake-menu-button Priorities 1`] = `
Array [
  <span
    className="fake-menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu"
      aria-live={null}
      className="btn fake-menu-button__button btn--primary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          Primary
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--dropdown"
          focusable={false}
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-dropdown"
          />
        </svg>
      </span>
    </button>
  </span>,
  <span
    className="fake-menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu"
      aria-live={null}
      className="btn fake-menu-button__button btn--tertiary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          Tertiary
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--dropdown"
          focusable={false}
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-dropdown"
          />
        </svg>
      </span>
    </button>
  </span>,
]
`;

exports[`Storyshots ebay-fake-menu-button Reverse (Menu grows to the left) 1`] = `
<div
  style={
    Object {
      "marginLeft": "100px",
    }
  }
>
  <span
    className="fake-menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-live={null}
      className="btn fake-menu-button__button btn--secondary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          Menu grows to the left
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--dropdown"
          focusable={false}
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-dropdown"
          />
        </svg>
      </span>
    </button>
  </span>
</div>
`;

exports[`Storyshots ebay-fake-menu-button Variants 1`] = `
Array [
  <h3>
    Button
  </h3>,
  <span
    className="fake-menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu inside the form"
      aria-live={null}
      className="btn fake-menu-button__button btn--secondary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          Button
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--dropdown"
          focusable={false}
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-dropdown"
          />
        </svg>
      </span>
    </button>
  </span>,
  <h3>
    Form
  </h3>,
  <span
    className="fake-menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu inside the form"
      aria-live={null}
      className="btn fake-menu-button__button btn--form"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          Form
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--dropdown"
          focusable={false}
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-dropdown"
          />
        </svg>
      </span>
    </button>
  </span>,
  <h3>
    Overflow
  </h3>,
  <span
    className="fake-menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu"
      className="fake-menu-button__button icon-btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon icon--overflow"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-overflow"
        />
      </svg>
    </button>
  </span>,
]
`;

exports[`Storyshots ebay-fake-menu-button With Custom Label 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span
          style={
            Object {
              "background": "url(https://ir.ebaystatic.com/pictures/aw/pics/cmp/ds3/sprds3_21.png)",
              "display": "inline-block",
              "height": "20px",
              "marginRight": "8px",
              "verticalAlign": "middle",
              "width": "26px",
            }
          }
        />
         Fun with flags!
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-fake-menu-button With Separator 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        Complex menu
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-fake-menu-button With icon 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <svg
          aria-hidden={true}
          className="icon icon--settings"
          focusable={false}
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-settings"
          />
        </svg>
        <span>
          Settings
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-fake-menu-button Without toggle icon 1`] = `
<span
  className="fake-menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn fake-menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Menu
  </button>
</span>
`;
